enclave/sgx/collateralinfo.c

```diff
@@ -169,6 +170,123 @@ oe_result_t oe_get_sgx_quote_verification_collateral(
     return result;
 }
 
+void oe_prealloc_quote_verification_collateral_args(
+oe_get_sgx_quote_verification_collateral_args_t* buf)
+{
+    /*These numbers are just estimates of buffer sizes*/
+    size_t prealloc_tcb_info_size = 5000;
+    size_t prealloc_tcb_info_issuer_chain_size = 3000;
+    size_t prealloc_pck_crl_size = 600;
+    size_t prealloc_root_ca_crl_size = 600;
+    size_t prealloc_pck_crl_issuer_chain_size = 3000;
+    size_t prealloc_qe_identity_issuer_chain_size = 3000;
+    size_t prealloc_qe_identity_size = 700;
+    if(!buf->tcb_info_size)
+    {
+        buf->tcb_info = (uint8_t *)malloc(prealloc_tcb_info_size);
+        if(buf->tcb_info != NULL)
+        {
+            buf->tcb_info_size = prealloc_tcb_info_size;
+            memset(buf->tcb_info,0, buf->tcb_info_size);   
```

Comment:
Instead of malloc + memset, [use calloc](http://www.cplusplus.com/reference/cstdlib/calloc/)